authentication.go

```diff
@@ -183,21 +155,23 @@ func createSignatureCommon(payload *models.AuthPayload, bodyString string, priva
 	}
 
 	// Signature, using bitcoin signMessage
-	var err error
-	if payload.Signature, err = bitcoin.SignMessage(
-		hex.EncodeToString(privateKey.Serialise()),
+	sigBytes, err := bsm.SignMessage(
+		privateKey,
 		getSigningMessage(key, payload),
-		true,
-	); err != nil {
+	)
+	if err != nil {
 		return nil, err
 	}
 
+	payload.Signature = base64.StdEncoding.EncodeToString(sigBytes)
+
 	return payload, nil
 }
 
-// getSigningMessage will build the signing message string
-func getSigningMessage(xPub string, auth *models.AuthPayload) string {
-	return fmt.Sprintf("%s%s%s%d", xPub, auth.AuthHash, auth.AuthNonce, auth.AuthTime)
+// getSigningMessage will build the signing message byte array
+func getSigningMessage(xPub string, auth *models.AuthPayload) []byte {
+	message := fmt.Sprintf("%s%s%s%d", xPub, auth.AuthHash, auth.AuthNonce, auth.AuthTime)
+	return []byte(message) // Convert string to byte array
```

Comment:
The comment seems to be redundant - I think every golang developer will understand what `[]byte(message)` does.

Reply:
removed